src/Traits/ComputesNanosecondTimestamps.php

```diff
@@ -17,22 +17,36 @@ public function getNanoSecondTimestamp($timestamp = null)
             return $timestamp->getTimestamp() * 1000000000;
         }
 
-        if (strlen($timestamp) == 19) {
+        $length = is_string($timestamp) ? strlen($timestamp) : 0;
```

Comment:
If I pass in a completely valid timestamp as an integer, you're going to set the length to 0 here because it isn't a string?